Use OrderedSet in _functorch/partitioners

masnesral opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #146102

In an attempt to make partitioning more deterministic, change all sets in partitioners.py to OrderedSets. Note that this change does not fix the non-determinism we're seeing in the internal model. But let's at least eliminate this potential source of non-determinism before investigating any changes to the mincut approach?

masnesral committed:
User OrderedSet in partitioners

Summary:

Test Plan:

[ghstack-poisoned]

masnesral committed:
Update on "Make partitioning more deterministic"


Copied from https://github.com/pytorch/pytorch/pull/145024, but made _all_ sets OrderedSets and fixed mypy complaints. Note that this change does not fix the non-determinism we're seeing in the internal model. But let's at least eliminate this potential source of non-determinism before investigating any changes to the mincut approach?

[ghstack-poisoned]

eellison reviewed (commented):
Can you add these files to the OrderedSet linter ? this will make the changes automatically, and update future set uses. ty @rec

masnesral committed:
Update on "Make partitioning more deterministic"


Copied from https://github.com/pytorch/pytorch/pull/145024, but made _all_ sets OrderedSets and fixed mypy complaints. Note that this change does not fix the non-determinism we're seeing in the internal model. But let's at least eliminate this potential source of non-determinism before investigating any changes to the mincut approach?

[ghstack-poisoned]

masnesral committed:
Update on "Make partitioning more deterministic"


Copied from https://github.com/pytorch/pytorch/pull/145024, but made _all_ sets OrderedSets and fixed mypy complaints. Note that this change does not fix the non-determinism we're seeing in the internal model. But let's at least eliminate this potential source of non-determinism before investigating any changes to the mincut approach?

[ghstack-poisoned]

oulgen commented on torch/_functorch/partitioners.py:
```diff
@@ -1375,13 +1400,13 @@ def get_default_op_list() -> OpTypes:
-        set(random_ops),
-        set(view_ops),
-        set(recomputable_ops),
+        OrderedSet(fusible_ops),
+        OrderedSet(compute_intensive_ops),
+        OrderedSet(random_ops),
+        OrderedSet(view_ops),
+        OrderedSet(recomputable_ops),
```
arent some of these already orderedset?

masnesral committed:
Update on "Make partitioning more deterministic"


Copied from https://github.com/pytorch/pytorch/pull/145024, but made _all_ sets OrderedSets and fixed mypy complaints. Note that this change does not fix the non-determinism we're seeing in the internal model. But let's at least eliminate this potential source of non-determinism before investigating any changes to the mincut approach?

[ghstack-poisoned]

Skylion007 commented on torch/_functorch/partitioners.py:
```diff
@@ -1879,12 +1910,12 @@ def classify_nodes(joint_module):
-            node.name for node in bw_module.graph.nodes if node.op == "call_function"
-        }
+        fw_module_nodes = OrderedSet(
+            [node.name for node in fw_module.graph.nodes if node.op == "call_function"]
```
These list comprehensions can be generators if you want an extra copy (input only needs to be an Iterable)

masnesral committed:
Update on "Make partitioning more deterministic"


Copied from https://github.com/pytorch/pytorch/pull/145024, but made _all_ sets OrderedSets and fixed mypy complaints. Note that this change does not fix the non-determinism we're seeing in the internal model. But let's at least eliminate this potential source of non-determinism before investigating any changes to the mincut approach?

[ghstack-poisoned]

masnesral committed:
Update on "Make partitioning more deterministic"


Copied from https://github.com/pytorch/pytorch/pull/145024, but made _all_ sets OrderedSets and fixed mypy complaints. Note that this change does not fix the non-determinism we're seeing in the internal model. But let's at least eliminate this potential source of non-determinism before investigating any changes to the mincut approach?

[ghstack-poisoned]

masnesral commented:
@pytorchbot merge
